src/zernike_fit.jl

```diff
@@ -1,33 +1,33 @@
-using FFTW, ZernikePolynomials, DSP
-include("pd_initialize.jl")
-include("suppzern.jl")
+# Fitting function zernike_img_fit()
+# Loss function zernikeloss()
+# Gradient function zernikegrad!()
+# Matrix preparation function loss_prep()
+#
 
-export zernike_img_fit
-
-function loss_prep!(Z, img, Hz, Zval)
+function loss_prep(Z, img, Hz, Zval)
     imsz = size(img)
-    Hk, Sk = ZernFF(Z, Hz, Zval, imsz)
+    Hk, Sk = ZernFT(Z, Hz, Zval, imsz)
 
     #compute transform of image
     Dk = fft(img)
 
     # penalty prep
-    S2tot = Sk .* conj(Sk)
+    S2tot = abs.(Sk .* conj(Sk))
```

Comment:
Could this be just `abs2.(Sk)`? I think that's faster (`a^2 + b^2` rather than `sqrt((a^2 + b^2) + (ab - ab)^2)` (which is the same, but more laboriously). That said, it's quite possible that LLVM would optimize all the extra operations away.

If you decide to change it here, there are several other places that the same change could be made.